ui/src/app/components/prepare-migration/prepare-migration.component.html

```diff
@@ -85,7 +101,27 @@ <h3>Target details:</h3>
                             </mat-icon>
                         </button></span>
                 </p>
-                <p class="point" *ngIf="(selectedMigrationType ==='lowdt') && !(selectedMigrationMode ==='Schema')">
+                <p class="point" *ngIf="(selectedMigrationType ==='lowdt') && !(selectedMigrationMode ==='Schema') && isSharded">
+                    <span class="bullet">2</span>
+                    <span>Setup Migration profile</span><span><button mat-button class="configure"
+                            (click)="openMigrationProfileForm()"
+                            [disabled]="isMigrationInProgress || !isTargetDetailSet">
+                            Configure
+                            <mat-icon>edit</mat-icon>
+                            <mat-icon *ngIf="isSourceConnectionProfileSet" class="success"
+                                matTooltip="Source connection profile configured" matTooltipPosition="above">
+                                check_circle
+                            </mat-icon>
+                        </button>
+                        <button mat-button class="configure"
+                        (click)="downloadConfiguration()"
+                        [disabled]="isMigrationInProgress || !isTargetDetailSet || !isTargetConnectionProfileSet">
+                            Download
```

Comment:
Yeah...its because the whole `sourceProfileConfig` object gets deserialised from object to JSON. This in unavoidable without implementing custom deserialisation. I'll log an enhancement bug for this for now - b/284232710

Reply:
I had to make some changes in the code flow to sort this out. Currently, Dataflow info was stored in the UI variables and passed as part of the input to the `migrate` API.
However, with the option to download JSON, we need the full configuration file to be available prior to the `migrate` API being actually called. 
I've added a new API which configures the Dataflow location information in the backend first now and updates the sharding configuration with it.
The `migrate` API for sharding will just pick the sharding configuration and perform the migration.

Thanks for the comment! ✌🏻